### PR TITLE
Let common library auto-install the app engine java component

### LINKED ISSUE #1048: app-engine-java component does not need to be preinstalled for cloud sdk dev server
Cloud SDK will install it automatically at run time if it's missing.

### PR DESCRIPTION
**Pull request by etanshaul**
fixes #1048

### CODE REVIEW

File: google-cloud-tools-plugin/ultimate/src/com/google/cloud/tools/intellij/appengine/server/instance/AppEngineServerModel.java [hunk 1]
```diff
@@ -174,11 +173,6 @@ public void checkConfiguration() throws RuntimeConfigurationException {
           .sdkPath(sdkService.getSdkHomePath())
           .build();
       sdk.validateCloudSdk();
```
Comment (joaoandremartins) [inline google-cloud-tools-plugin/ultimate/src/com/google/cloud/tools/intellij/appengine/server/instance/AppEngineServerModel.java]: We can probably go back to not instantiating the sdk variable and calling validateCloudSdk() after build().

File: google-cloud-tools-plugin/ultimate/src/com/google/cloud/tools/intellij/appengine/server/run/CloudSdkStartupPolicy.java [hunk 2]
```diff
@@ -78,11 +77,6 @@ public OSProcessHandler createProcessHandler(
                   .sdkPath(sdkService.getSdkHomePath())
                   .build();
               sdk.validateCloudSdk();
-              sdk.validateAppEngineJavaComponents();
```
Comment (joaoandremartins) [inline google-cloud-tools-plugin/ultimate/src/com/google/cloud/tools/intellij/appengine/server/run/CloudSdkStartupPolicy.java]: As above.

File: google-cloud-tools-plugin/ultimate/src/com/google/cloud/tools/intellij/appengine/server/instance/AppEngineServerModel.java (hunk 1 above)
Comment (etanshaul) [inline google-cloud-tools-plugin/ultimate/src/com/google/cloud/tools/intellij/appengine/server/instance/AppEngineServerModel.java]: sgtm

**Comment by patflynn** [comment]
@etanshaul 
can we merge this?

**Comment by joaoandremartins** [comment]
There were a couple of very minor comments from me that I don't see addressed yet.

**Comment by etanshaul** [comment]
@joaoandremartins which are still unaddressed? I thought I had covered them all.

**Comment by joaoandremartins** [comment]
Sorry, github fail here. Clicked "view changes" on my comment and obviously that brought me to an older commit...

**Pull request merged**